helm/thingsboard/templates/mqtt-transport.yaml

```diff
@@ -0,0 +1,139 @@
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.mqtt.securityContext | nindent 12 }}
+          image: "{{ .Values.mqtt.image.server | default .Values.global.image.server }}/{{ .Values.mqtt.image.repository }}:{{ .Values.mqtt.image.tag | default .Values.global.image.tag }}"
+          imagePullPolicy: {{ .Values.mqtt.image.pullPolicy | default .Values.global.image.pullPolicy}}
+          ports:
+          - containerPort: {{ .Values.mqtt.port.number }}
+            name: {{ .Values.mqtt.port.name }}
+          env:
+          - name: TB_SERVICE_ID
+            value: "tb-mqtt-transport"
+          - name: TB_QUEUE_TYPE
+            value: "kafka"
+          - name: MQTT_TIMEOUT
+            value: "{{ .Values.mqtt.timeout }}"
+          - name: TB_KAFKA_SERVERS
+            value: "{{ .Release.Name }}-kafka-headless:9092"
+          {{- if eq .Values.mqtt.ssl.enabled true }}
+          - name: MQTT_SSL_ENABLED
+            value: "true"
+          - name: MQTT_SSL_CREDENTIALS_TYPE
+            value: "PEM"
+          - name: MQTT_SSL_PEM_CERT
+            value: /mqtt-certs/mqttserver.pem
+          - name: MQTT_SSL_PEM_KEY
+            value: /mqtt-certs/mqttserver_key.pem
+          - name: MQTT_SSL_BIND_ADDRESS
+            value: "0.0.0.0"
+          - name: MQTT_SSL_BIND_PORT
+            value: "{{ .Values.mqtt.port.number }}"
+          {{- else }}
+          - name: MQTT_BIND_ADDRESS
+            value: "0.0.0.0"
+          - name: MQTT_BIND_PORT
+            value: "{{ .Values.mqtt.port.number }}"
+          {{- end }}
+          volumeMounts:
+            - mountPath: /config
+              name: {{ .Release.Name }}-mqtt-transport-config
+          {{- if eq .Values.mqtt.ssl.enabled true }}
+            - mountPath: /mqtt-certs
+              name: mqtt-certs
+              readOnly: true
+          {{- end }}
+          readinessProbe:
+            periodSeconds: 20
+            tcpSocket:
+              port: {{ .Values.mqtt.port.number }}
+          livenessProbe:
+            initialDelaySeconds: 120
+            periodSeconds: 20
+            tcpSocket:
+              port: {{ .Values.mqtt.port.number }}
+          resources:
+            {{- toYaml .Values.mqtt.resources | nindent 12 }}
+      volumes:
+        - name: {{ .Release.Name }}-mqtt-transport-config
+          configMap:
+            name: {{ .Release.Name }}-mqtt-transport-config
+            items:
+            - key: conf
+              path:  tb-mqtt-transport.conf
+            - key: logback
+              path:  logback.xml
+        - name: {{ .Release.Name }}-node-logs
+          emptyDir: {}
+        {{- if eq .Values.mqtt.ssl.enabled true }}
+        - name: mqtt-certs
+          secret:
+            defaultMode: 420
+            items:
+            - key: {{ .Values.mqtt.ssl.cert }}
+              path: mqttserver.pem
+            - key: {{ .Values.mqtt.ssl.key }}
```

Comment:
When a kubernetes.io/tls type of secret is created in k8s, it must contain tls.crt & tls.key keys. So, perhaps these should not be optional values at all?

Reply:
OK!